source/blog/2022-11-17-announcing-hanami-200rc.html.markdown

```diff
@@ -0,0 +1,357 @@
+
+### Type-safe app settings
+
+For your app to do its thing, you'll want to give it various settings: behavioral toggles, API keys, external system URLs and the like. Hanami 2.0 provides a built-in settings class along with [dotenv][dotenv] integration for loading these settings from `.env*` files in local development.
+
+You can define your settings along with type constructors in `config/settings.rb`:
+
+```ruby
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :email_client_api_key, constructor: Types::String
+    setting :emails_enabled, constructor: Types::Params::Bool
+  end
+end
+```
+
+Your settings are then available as a `"settings"` component for you to use from anywhere in your app:
+
+```ruby
+# Given ENV["EMAILS_ENABLED"] = "true"
+Hanami.app["settings"].emails_enabled # => true
+```
+
+This means you can include `"settings"` in a list of `Deps` and access your settings from any class:
+
+```ruby
+module Bookshelf
+  module Emails
+    class DailyUpdate
+      include Deps["email_client", "settings"]
+
+      def deliver(recipient)
+        return unless settings.emails_enabled
+        # ...
+      end
+    end
+  end
+end
+```
+
+Hanami also loads your settings early in app boot and will raise an error for any settings that fail to meet their type expectations, giving you early feedback and ensuring your app doesn’t boot in a potentially invalid state.
+
+[dotenv]: https://github.com/bkeepers/dotenv
+
+### Providers
+
+So far we’ve seen how your classes and settings can be accessed as components in your app. But what about components that require special handling to initialize? For these, Hanami 2.0 introduces providers. Here’s one for the `"email_client"` component we used earlier, in `config/providers/email_client.rb`:
+
+```ruby
+Hanami.app.register_provider :email_client do
+  prepare do
+    require "acme_email/client"
+  end
+
+  start do
+    client = AcmeEmail::Client.new(
+      api_key: target["settings"].email_client_api_key,
+      default_from: "no-reply@bookshelf.example.com"
+    )
+
+    register "email_client", client
+  end
+end
+```
+
+Every provider has its own set of `prepare`/`start`/`stop` lifecycle steps, which is useful for setting up and using heavyweight components, or components that use tangible resources like database connections.
+
+### Slices
+
+As your app grows up, you may want to draw clear boundaries between its major areas of concern. For this, Hanami 2.0 introduces slices, a built-in facility for organising your code, encouraging maintainable boundaries, and creating operational flexibility.
+
+To get started with a slice, create a directory under `slices/` and then start adding your code there, using a matching Ruby namespace. It’s that simple! Here’s a class that imports books for our bookshelf app:
+
+```ruby
```

Comment:
```suggestion
```ruby
# slices/feeds/lib/process_feed.rb
```

Reply:
Good idea. It's actually `slices/feeds/process_feed.rb`, even simpler!!